pkg/registrar/registrar.go

```diff
@@ -0,0 +1,91 @@
+/*
```

Comment:
OK, after thinking about this more. I found it would still be helpful to maintain bi-direction mapping. Will address all above comments.